### PR TITLE
ci: use dependency review action

### PR DESCRIPTION
**Pull request by HonkingGoose**
## Changes

- Use [`actions/dependency-review-action`](https://github.com/actions/dependency-review-action) to avoid vulnerable dependencies

## Context

The Dependency Review GitHub Action is currently in public beta and subject to change.

There's a [GitHub Blog](https://github.blog/2022-04-06-prevent-introduction-known-vulnerabilities-into-your-code/) which explains the use case.

Read the [GitHub Docs, Dependency review enforcement](https://docs.github.com/en/code-security/supply-chain-security/understanding-your-software-supply-chain/about-dependency-review#dependency-review-enforcement) and check the [`actions/dependency-review-action`](https://github.com/actions/dependency-review-action) code.

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by viceice — COMMENTED** [review]
any particular issue to solve? 🤔

**Comment by HonkingGoose** [comment]
@viceice I think this [GitHub Blog](https://github.blog/2022-04-06-prevent-introduction-known-vulnerabilities-into-your-code/) explains the use case the best.

The big idea is that the action errors out when a PR introduces a vulnerable dependency. This way we don't accidentally add a dependency with known vulnerabilities.

But maybe we don't need this because we cover this with some other tool already.

---

**Comment by JamieMagee** [comment]
I like the idea, and I think we should test it out.

Renovate offers similar functionality by opening PRs with security fixes, but this action has the ability to break builds.

**Review by JamieMagee — APPROVED** [review]

File: .github/workflows/dependency-review.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+jobs:
+  dependency-review:
+    runs-on: ubuntu-latest
+    steps:
+      - name: 'Checkout Repository'
+        uses: actions/checkout@a12a3943b4bdde767164f792f33f40b04645d846 # tag=v3.0.0
+
+      - name: 'Dependency Review'
+        uses: actions/dependency-review-action@3f943b86c9a289f4e632c632695e2e0898d9d67d # tag=v1
```
Comment (viceice) [inline .github/workflows/dependency-review.yml:15]: don't we have a more explicit version? 😕

**Review by viceice — COMMENTED** [review]
otherwise LGTM

File: .github/workflows/dependency-review.yml (hunk 1 above)
Comment (JamieMagee) [reply .github/workflows/dependency-review.yml:15]: Unfortunately not. The only tag offered right now is v1[^1]. I've opened an issue upstream[^2]

[^1]: https://github.com/actions/dependency-review-action/tags
[^2]: https://github.com/actions/dependency-review-action/issues/21

**Review by viceice — APPROVED** [review]

**Pull request merged**